pkg/http/listener.go

```diff
@@ -182,10 +192,12 @@ func (listener *httpListener) start() {
 			// Peek bytes of maximum length of all HTTP methods.
 			data, err := bufconn.Peek(methodMaxLen)
 			if err != nil {
-				if listener.errorLogFunc != nil {
-					listener.errorLogFunc(err,
-						"Error in reading from new TLS connection %s at server %s",
-						bufconn.RemoteAddr(), bufconn.LocalAddr())
+				if !isRoutineNetErr(err) {
+					if listener.errorLogFunc != nil {
```

Comment:
You could do `if listener.errorLogFunc != nil && !ignoreErr(err)` to avoid necessary nesting `if`

Reply:
Done